Dynamic/FormFieldTypePool.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Sulu\Bundle\FormBundle\Dynamic;
+
+class FormFieldTypePool
+{
+    /**
+     * @var FormFieldTypeInterface[]
+     */
+    private $types;
+
+    public function __construct()
```

Comment:
add argument for types they will be passed by the compiler pass. then you can remove the add function